scripts/e2e-tests.sh

```diff
@@ -197,23 +232,23 @@ function create_test_cluster() {
   # Don't fail test for kubetest, as it might incorrectly report test failure
   # if teardown fails (for details, see success() below)
   set +o errexit
-  run_go_tool k8s.io/test-infra/kubetest \
+  { run_go_tool k8s.io/test-infra/kubetest \
```

Comment:
For retrying, encapsulate only the call to kubetest, instead of the whole function; it's simpler and avoids an even more convoluted log.

Reply:
Not sure how to achieve that, could you elaborate more? @adrcunha

Reply:
You made create_test_cluster a helper function, and you run it entirely if cluster creation fails. Instead, extract the kubetest part to a separate function, and only rerun such function if cluster creation fails.

Reply:
Done. Please review again @adrcunha

Reply:
You're still encapsulating unnecessary code/initialization/teardown (like the setup hook), and factoring out constant parameters (like the test command and up/down flags). Let's sync up in person about this.

Reply:
Updated the PR based on what we discussed in person, @adrcunha please review when you got a chance